packages/validator-rules/lib/loadRules.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {oneBehindFetch} = require('@ampproject/toolbox-core');
+
+const VALIDATOR_RULES_URL = 'https://cdn.ampproject.org/v0/validator.json';
+const VALIDATOR_RULES_LOCAL = '../../validator.json';
+
+const isNodeJs = typeof process !== 'undefined';
+
+async function loadRemote(url) {
+  const req = await oneBehindFetch(url);
+  return req.json();
+}
+
+async function loadLocal() {
+  if (!isNodeJs) {
+    throw new Error('Local loading is not supported in browsers');
+  }
+
+  const {promisify} = require('util');
+  const readFileAsync = promisify(require('fs').readFile);
+
+  const data = await readFileAsync(getLocalPath());
+  return JSON.parse(data);
+}
+
+async function loadDefault(url) {
+  if (!isNodeJs) {
+    return loadRemote(url);
+  }
+
+  const {existsSync} = require('fs');
+
+  if (existsSync(getLocalPath())) {
+    return loadLocal();
+  }
+  return loadRemote(url);
+}
+
+function getLocalPath() {
+  const path = require('path');
+  return path.join(__dirname, VALIDATOR_RULES_LOCAL);
+}
+
+async function loadRules({source, url}) {
+  url = url || VALIDATOR_RULES_URL;
+  switch (source) {
+    case 'local':
```

Comment:
Instead of configuring this via options, you could also provide to methods `fetch` and `loadFile` on the ValidationRules object.

Reply:
I feel the functionality is not different enough to vouch having two different functions and that also prevents me from easily having the default option which is "try local if possible".

That said, maybe `source` is not needed if the `url` is a local file, i.e. use something like https://github.com/ampproject/amp-toolbox/blob/533464d9a198f0addbc0f79c8b213934fd9b32d3/packages/cli/lib/io.js#L25

Ideally, whatever the resolution of #378 ends up being will replace this file anyway.

Reply:
Sounds good. I'd suggest removing loadLocal until we settled on a common approach. Once it's out there it's harder to deprecate and I don't think that the feature is critical.

Reply:
Fair point, done.